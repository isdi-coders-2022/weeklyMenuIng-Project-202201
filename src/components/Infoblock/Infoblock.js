import styledComponents from "styled-components";
import PropTypes from "prop-types";

const Text = styledComponents.h1`
color : #151f28;
text-align: center;
background-color: #00ABBB;
margin: 0;
padding: 20px;
`;

const Infoblock = ({ text }) => {
  return <Text>{text}</Text>;
};

Infoblock.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Infoblock;
