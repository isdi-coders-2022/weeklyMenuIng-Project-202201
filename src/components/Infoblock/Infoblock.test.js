import { render, screen } from "@testing-library/react";
import Infoblock from "./Infoblock";

describe("Given a Infoblock component", () => {
  describe("When instantiated", () => {
    test("It should render a heading", () => {
      render(<Infoblock text="" />);

      const heading = screen.queryByRole("heading");

      expect(heading).toBeInTheDocument();
    });
  });

  describe("When instantiated with text 'Welcome to My Weekly Menu'", () => {
    test("It should render a heading with text 'Welcome to My Weekly Menu'", () => {
      const text = "Welcome to My Weekly Menu";
      render(<Infoblock text={text} />);

      const displayedText = screen.queryByText(text);

      expect(displayedText).toBeInTheDocument();
    });
  });
});
