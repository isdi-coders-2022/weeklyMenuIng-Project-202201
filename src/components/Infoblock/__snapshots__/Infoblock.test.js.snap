// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Infoblock component When it's rendered Then it should match the snapshot 1`] = `
<h1
  className="sc-bdvvtL kVWPwq"
/>
`;
